src/openrct2/actions/LandSetHeightAction.hpp

```diff
@@ -0,0 +1,334 @@
+                    {
+                        rct_ride_entry* rideEntry = get_ride_entry_by_ride(ride);
+                        if (rideEntry != nullptr)
+                        {
+                            int32_t maxHeight = rideEntry->max_height;
+                            if (maxHeight == 0)
+                            {
+                                maxHeight = RideData5[get_ride(rideIndex)->type].max_height;
+                            }
+                            int32_t zDelta = tileElement->clearance_height - _height;
+                            if (zDelta >= 0 && zDelta / 2 > maxHeight)
+                            {
+                                return std::make_unique<GameActionResult>(GA_ERROR::DISALLOWED, STR_SUPPORTS_CANT_BE_EXTENDED);
+                            }
+                        }
+                    }
+                }
+            } while (!(tileElement++)->IsLastForTile());
+        }
+
+        uint8_t zCorner = _height; // z position of highest corner of tile
+        TileElement* surfaceElement = map_get_surface_element_at(_coords);
+        if (surfaceElement->AsSurface()->HasTrackThatNeedsWater())
+        {
+            uint32_t waterHeight = surfaceElement->AsSurface()->GetWaterHeight();
+            if (waterHeight != 0)
+            {
+                if (_style & 0x1F)
+                {
+                    zCorner += 2;
+                    if (_style & 0x10)
+                    {
+                        zCorner += 2;
+                    }
+                }
+                if (zCorner > waterHeight * 2 - 2)
+                {
+                    surfaceElement++;
+                    map_obstruction_set_error_text(surfaceElement);
+                    return std::make_unique<GameActionResult>(GA_ERROR::DISALLOWED, gGameCommandErrorText);
+                }
+            }
+        }
+
+        zCorner = _height;
+        if (_style & 0xF)
+        {
+            zCorner += 2;
+            if (_style & 0x10)
+            {
+                zCorner += 2;
+            }
+        }
+
+        if (!gCheatsDisableClearanceChecks)
+        {
+            if (!map_can_construct_with_clear_at(
+                    _coords.x, _coords.y, _height, zCorner, &map_set_land_height_clear_func, 0xF, 0, nullptr,
+                    CREATE_CROSSING_MODE_NONE))
+            {
+                return std::make_unique<GameActionResult>(GA_ERROR::DISALLOWED, STR_NONE);
+            }
+        }
+
+        if (!gCheatsDisableClearanceChecks)
+        {
+            TileElement* tileElement = map_get_first_element_at(_coords.x / 32, _coords.y / 32);
+            do
+            {
+                int32_t elementType = tileElement->GetType();
+
+                if (elementType == TILE_ELEMENT_TYPE_WALL)
+                    continue;
+                if (elementType == TILE_ELEMENT_TYPE_SMALL_SCENERY)
+                    continue;
+                if (tileElement->flags & 0x10)
```

Comment:
Should be replaced with the appropriate constant.

Reply:
It should, but this is moved code, not new code. Not blocking IMO.

Reply:
For me, it is. It's trivial to fix.